build.gradle

```diff
@@ -163,42 +163,58 @@ task wrapper(type: Wrapper) {
-    onlyIfNewer true
-}
-
-
-task updateHoverflyBinaries(dependsOn: downloadHoverflyDistributions) {
-    def distDir = file(downloadHoverflyDistributions.dest)
-    def extractBinary = { FileTree zip ->
-        copy {
-            from zip
-            into 'src/main/resources/binaries'
-            rename { String filename ->
-                def p1 = Pattern.compile('hoverfly(.*)')
-                def m1 = p1.matcher(filename)
-                def extension = m1.find() ? m1.group(1) : ''
-                def p2 = Pattern.compile('hoverfly_bundle(.+?)\\.zip')
-                def m2 = p2.matcher(zip.asPath)
-                def type = m2.find() ? m2.group(1) : ''
-                filename = 'hoverfly' + type + extension
+task (updateHoverflyBinaries) {
+    doLast {
+        def downloadUrl = 'https://github.com/SpectoLabs/hoverfly/releases/download'
+        def version = 'v0.16.0'
+        def downloadDir = "$buildDir/tmp/dist-${version}"
+        println "Downloading Hoverfly $version if needed"
+        download {
+            src([
+                    "$downloadUrl/$version/hoverfly_bundle_linux_386.zip",
+                    "$downloadUrl/$version/hoverfly_bundle_linux_amd64.zip",
+                    "$downloadUrl/$version/hoverfly_bundle_OSX_amd64.zip",
+                    "$downloadUrl/$version/hoverfly_bundle_windows_386.zip",
+                    "$downloadUrl/$version/hoverfly_bundle_windows_amd64.zip"
+            ])
+            dest "$downloadDir"
+            onlyIfNewer true
+            quiet true
+        }
+        println "Unzipping Hoverfly ${version} binaries"
+
+        def distDir = file("$downloadDir")
+        def extractBinary = { FileTree zip ->
+            copy {
+                from zip
+                into 'src/main/resources/binaries'
+                rename { String filename ->
+                    def p1 = Pattern.compile('hoverfly(.*)')
+                    def m1 = p1.matcher(filename)
+                    def extension = m1.find() ? m1.group(1) : ''
+                    def p2 = Pattern.compile('hoverfly_bundle(.+?)\\.zip')
+                    def m2 = p2.matcher(zip.asPath)
+                    def type = m2.find() ? m2.group(1) : ''
+                    filename = 'hoverfly' + type + extension
+                }
             }
         }
+
+        distDir.listFiles()
+                .findAll { it.name.endsWith('.zip') }
+                .collect { zipTree(it).matching { include 'hoverfly*' } }
+                .each { zip -> extractBinary(zip) }
     }
+}
+
+task (cleanHoverflyBinaries, type: Delete) {
+    delete fileTree(dir: 'src/main/resources/binaries')
+}
 
-    distDir.listFiles()
-            .findAll { it.name.endsWith('.zip') }
-            .collect { zipTree(it).matching { include 'hoverfly*' } }
-            .each { zip -> extractBinary(zip) }
-}
\ No newline at end of file
+processResources {
+    dependsOn updateHoverflyBinaries
+}
+
+clean {
+    dependsOn cleanHoverflyBinaries
```

Comment:
We should let developers decide when they clean the binaries. If someone accidentally run `clean`, they will have to redownload everything again.